File: Week 07/id_011/reverse-bits.go
package algorithm00401

func reverseBits(num uint32) uint32 {
	bits, count := uint32(0), uint32(0)
	for count > 0 {
		bits |= (nums & 1) << (count - 1)
		num >>= 1
		count--
	}
	return bits
}

File: Week 07/id_081/ReverseBits.java
/**
 * ReverseBits
 */
public class ReverseBits {

    public int reverseBits(int n) {
        int result = 0;
        for(int i = 0;i < 32;i++){
            result = (result << 1) + (n & 1);
            n >>= 1;
        }
        return result;
    }
}